spot-vaults/contracts/_interfaces/ProtocolErrors.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+/// @notice Expected contract call to be triggered by authorized caller.
+error UnauthorizedCall();
+
+/// @notice Expected interface contract to return a fixed point with a different number of decimals.
+error UnexpectedDecimals();
+
+/// @notice Expected perc value to be at most (1 * 10**DECIMALS), i.e) 1.0 or 100%.
+error InvalidPerc();
+
+/// @notice Expected Senior CDR bound to be more than 1.0 or 100%.
+error InvalidSeniorCDRBound();
+
+/// @notice Expect AR lower bound to be under 1.0 and upper bound to be above 1.0.
+error InvalidARBound();
+
+/// @notice Expected pre and post swap AR delta to be non-increasing or non-decreasing.
+error UnexpectedARDelta();
```

Comment:
I prefer breaking errors out from the main contract file, mostly because it make the file smaller. We could call these `BillBrokerErrors` or `VaultErrors` rather than `ProtocolErrors`?

Reply:
I think the name can be whatever you think is best. The most important piece is moving it so it's clear it's "outside" the core protocol and next to the vault implementation wherever that is.